### ISSUE TITLE
Install package not available

### ISSUE THREAD

**Issue by irispils** [envato/wp-envato-market#74]
Hello!
I just installed the plugin and linked both Global OAuth Personal Token and the themes I'm interested in. All good so far. When I go back to the "Themes" tab and try to install the themes, I get this error:
"Install package not available"

Any idea why? Any help really appreciated!

**Comment by JosKlever** [comment]
That might be an issue of the specific theme. And I suppose you have purchased the themes?

**Comment by irispils** [comment]
Yes I purchased the themes and it is several by different companies. I guess that all can have the same problem, but I wanted to make sure it's not something else. Is there any way I can verify that it is those themes?

**Comment by JosKlever** [comment]
You can google for the error message combined with the theme name, create a ticket for the theme maker, etc. It looks like the error message isn't coming from the Envato Market plugin, but from Themeforest/Envato.
I'm using Envato Market on various websites with combinations of themes and hosting providers. It are popular themes however, so that might be the reason I haven't encountered this issue myself yet.

**Comment by JorgeDDW** [comment]
I got this problem a few minutes ago. You can solve this issues with the Api Key permission on https://build.envato.com/my-apps.

**Comment by acseven** [comment]
I'm having this issue when using a token for a specific item ID, and even with all the token's permissions set it still returns the same download not available error. 

My basic settings were as mentioned [here](https://github.com/envato/wp-envato-market/issues/53#issuecomment-230916320), but again, even with all of them selected it doesn't make a difference.

**Comment by beer73** [comment]
Any solution to this issue? I'm in same boat.
With the Global OAuth Token (all themes are shown = bad) works fine (themes are installed ok) but with Single Use OAuth Token activated only for one item (only the theme activated are shown = good!) the same error happens "Install package not available" and can't install the theme. I tried playing with all Api Key permissions check marks and nothing change.

**Comment by valendesigns** [comment]
Even though a theme is showing it's not guaranteed to be installable. The reason is that if the author has not uploaded an installable theme archive the API will fail to fetch the resource, but without any reason why beyond that it failed. So then it just shows in the plugin there was a failure but nothing about what failed. This is where the confusion happens. It's difficult to error handle the API.

**Comment by beer73** [comment]
Hi Derek, thanks for your work. 
I think the theme author has effectively uploaded the installable archive, because it installs ok always with Global Token but for some reason it always fail with Single Token, so that does not seem to be the problem.

I take some time and I tested it with the following 8 themes, all install ok with Global Token but fails with Single Token:

border 6920241
enfold 4519990
division 5030589
titan 5072056
nevia 4118819
midway 3559006
dandelion 136628
the landscaper 13460357

For your comments, maybe the problem may be in the Envato side, I dont know.

My personal token have the following default permissions:

- View and search Envato sites
- Download your purchased items
- Verify purchases of your items
- List purchases you've made
- Verify purchases you've made

However I have also tried it with all the active checks.

**Comment by GabrielAPass** [comment]
Thanks for pointing out the solution to use a Global OAuth token.  I wouldn't have thought to try that.

Clearly this is an issue with the API.  For the record, another theme that fails with a Single Use token but works with Global is:

[Safeguard 14781353](https://themeforest.net/item/safeguard-security-services-wordpress-theme/14781353)

**Comment by drugurkocak** [comment]
Hi,
I purchased Envision theme for Wordpress and tried to install using global and single use tokens.
I remember I succeeded once using Single Use OAuth Personal Tokens. Today, I can see the theme name under the Themes tab, but I get the same error "package not available" when I try to install via Envato Market plugin. 
[Envision](https://themeforest.net/item/envision-responsive-retina-multipurpose-theme/5928018)
